### ISSUE TITLE
Empty In-Masks and PnP Error

### ISSUE THREAD

**Issue by zr-zhou0o0** [520xyxyzq/3DGS-CD#9]
Dear Author,

I encountered several bugs during the execution of the code. And the in-mask prediction fails, all in-masks are empty. Here are the issues I faced (and how I tried to fix them):

1. After setting up the environment, I reinstalled the gsplat library with v0.1.11. The version of gsplat was not specified, but the API underwent a significant update in v1.0.0.

2. The version of pycolmap was also unspecified. In pycolmap 3.11, the `absolute_pose_estimation` function was renamed to `estimate_and_refine_absolute_pose`. So I downgraded pycolmap to version 3.10.0.

3. Regardless of the pycolmap version I used, the 'success' attribute was missing. In the `PnP` function in file `nerfstudio/utils/obj_3d_seg.py`, I commented out these lines:
   ```
   if not ret['success']:
       print("Warn: PnP failed!!")
       return None, 0.0, 0
   ```

4.The `qvec` and `tvec` attributes were missing too. I modified these lines:
   ```
   R_mat = torch.tensor(Quaternion(*ret['qvec']).rotation_matrix)
   tvec = torch.tensor(ret['tvec'])
   ```
   to
   ```
   rigid_transform = ret['cam_from_world']
   R_mat = torch.tensor(Quaternion(rigid_transform.rotation.quat).rotation_matrix)
   tvec = torch.tensor(rigid_transform.translation)
   ```

No other code has been modified. Here are the [debug](https://drive.google.com/file/d/1tTAZg1MV4rnqE_c1uKnFdI7eNSk3Jh9M/view?usp=sharing) and the [masks_new](https://github.com/user-attachments/files/18885879/masks_new.zip). Noticing that the gaussians are empty in the debug depths, I suspect the error is due to pose estimation. However both `qvec` and `tvec` seem to be correct.
I would appreciate your help with this issue.

**Comment by 520xyxyzq** [comment]
Dear Zirui,

Thanks very much for the feedback. I've updated the Installation section to have the changes needed. I used `pycolmap 0.4.0` (and `gsplat 0.1.9`). Could you try running the code again with this version?

For the move-in mask issue, it does seem like a pose estimation problem (since move-out masks worked). If the version fix doesn’t resolve it, could you share the `R_mat` and `tvec` from your run?

Thanks!

**Comment by zr-zhou0o0** [comment]
Thanks for your help! I updated to pycolmap 0.4.0 and gsplat 0.1.9, and the issue is resolved.
